lib/matplotlib/colorbar.py

```diff
@@ -488,6 +488,16 @@ def __init__(self, ax, mappable=None, *, cmap=None,
         if isinstance(mappable, contour.ContourSet) and not mappable.filled:
             self.add_lines(mappable)
 
+        # Link the Axes and Colorbar for interactive use
+        self.ax._colorbar = self
```

Comment:
There is already a `ax._colorbar_info` that should identify an axes as a colorbar.

Reply:
I think the only thing I used this for is knowing the orientation, which isn't currently in the `_colorbar_info`. I could add that in there as another quantity... Would that be able to get out of sync though if someone passes their own axes into Colorbar, or is that considered robust to check if an axes is used for a colorbar? (It looks like currently someone can pass their own axes into `matplotlib.colorbar.Colorbar(ax, orientation='horizontal')` and ax wouldn't have _colorbarinfo unless it was previously created with the make_colorbar_axes functions?

Reply:
No you are right, thats not there is the user adds the axes.  

I guess this is OK - given that this and recent PRs are passing info back and forth between the axes and the colorbar a fair bit I do wonder if there is a better way to e doing this.  I appreciate your idea for making the colorbar a special type of axes now.